libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -346,15 +381,30 @@ else if (record.getBoolean(LOCALLY_CREATED))
             action = Action.create;
         else if (record.getBoolean(LOCALLY_UPDATED))
             action = Action.update;
-
         if (action == null) {
+
             // Nothing to do for this record
             return true;
         }
 
         // Getting type and id
-        String objectType = (String) SmartStore.project(record, Constants.SOBJECT_TYPE);
-        String objectId = record.getString(Constants.ID);
+        final String objectType = (String) SmartStore.project(record, Constants.SOBJECT_TYPE);
+        final String objectId = record.getString(Constants.ID);
+        final long lastModifiedDate = record.optLong(SmartStore.SOUP_LAST_MODIFIED_DATE, UNCHANGED);
```

Comment:
Looks like I forgot to commit some local changes. There's a bug here with the last mod date. I'll add the fix in my next commit.